pkg/reconciler/common/transformer_additional_options.go

```diff
@@ -135,6 +135,66 @@ func (ot *OptionsTransformer) updateAnnotations(u *unstructured.Unstructured, an
 	return nil
 }
 
+func (ot *OptionsTransformer) updatePodTemplateLabels(u *unstructured.Unstructured, labels map[string]string) error {
+	if len(labels) == 0 {
+		return nil
+	}
+
+	actualLabels, _, err := unstructured.NestedMap(u.Object, "spec", "template", "metadata", "labels")
+
+	if err != nil {
+		return err
+	}
+
+	if actualLabels == nil {
+		actualLabels = make(map[string]interface{})
+	}
+
+	for labelKey, labelValue := range labels {
+		actualLabels[labelKey] = labelValue
+	}
+
+	if actualLabels == nil {
+		unstructured.RemoveNestedField(u.Object, "spec", "template", "metadata", "labels")
+	}
+
+	if err := unstructured.SetNestedMap(u.Object, actualLabels, "spec", "template", "metadata", "labels"); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (ot *OptionsTransformer) updatePodTemplateAnnotations(u *unstructured.Unstructured, annotations map[string]string) error {
+	if len(annotations) == 0 {
+		return nil
+	}
+
+	actualAnnotations, _, err := unstructured.NestedMap(u.Object, "spec", "template", "metadata", "annotations")
+
+	if err != nil {
+		return err
+	}
+
+	if actualAnnotations == nil {
+		actualAnnotations = make(map[string]interface{})
+	}
+
+	for annotationKey, annotationValue := range annotations {
+		actualAnnotations[annotationKey] = annotationValue
+	}
+
+	if actualAnnotations == nil {
```

Comment:
@charlesmorin `actualAnnotations` never be `nil` right? You have already initialized with empty map, if it is `nil`